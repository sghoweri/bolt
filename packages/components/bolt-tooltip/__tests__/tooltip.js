import {
  isConnected,
  render,
  renderString,
  stopServer,
  html,
} from '../../../../packages-extras/testing/testing-helpers';
import schema from '../tooltip.schema';
const { placement, uuid } = schema.properties;
const timeout = 120000;

describe('<bolt-tooltip> Component', () => {
  afterAll(async () => {
    await stopServer();
  }, 100);

  // Basic Usage
  test('Basic usage', async () => {
    const results = await render('@bolt-components-tooltip/tooltip.twig', {
      trigger: 'CRM',
      content: 'Customer relationship management',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  // Advanced Usage
  test('Advanced usage: adding tooltip to button', async () => {
    const results = await render('@bolt-components-tooltip/tooltip.twig', {
      trigger: '<bolt-button>Download</bolt-button>',
      content: 'File size: 25MB',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  // Props
  placement.enum.forEach(async placementChoice => {
    test(`Placement of the tooltip bubble: ${placementChoice}`, async () => {
      const results = await render('@bolt-components-tooltip/tooltip.twig', {
        trigger: 'CRM',
        content: 'Customer relationship management',
        placement: placementChoice,
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });

  test(`UUID of the tooltip`, async () => {
    const results = await render('@bolt-components-tooltip/tooltip.twig', {
      trigger: 'CRM',
      content: 'Customer relationship management',
      uuid: 'custom-unique-id',
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });
});
