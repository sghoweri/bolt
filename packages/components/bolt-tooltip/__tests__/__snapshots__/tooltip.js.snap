// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-tooltip> Component Advanced usage: adding tooltip to button 1`] = `
<bolt-tooltip uuid="12345">
  <div class="c-bolt-tooltip c-bolt-tooltip--top  ">
    <div role="button"
         tabindex="0"
         aria-describedby="js-bolt-tooltip-12345"
         aria-controls="js-bolt-tooltip-12345"
    >
      <ssr-keep for="bolt-tooltip">
        <bolt-button>
          Download
        </bolt-button>
      </ssr-keep>
    </div>
    <div id="js-bolt-tooltip-12345"
         class="c-bolt-tooltip__content"
         role="tooltip"
         aria-hidden="true"
    >
      <span class="c-bolt-tooltip__bubble">
        <ssr-keep for="bolt-tooltip">
          <span slot="content">
            File size: 25MB
          </span>
        </ssr-keep>
      </span>
    </div>
  </div>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Basic usage 1`] = `
<bolt-tooltip uuid="12345">
  <div class="c-bolt-tooltip c-bolt-tooltip--top c-bolt-tooltip--text-wrap c-bolt-tooltip--text-align-center">
    <div role="button"
         tabindex="0"
         aria-describedby="js-bolt-tooltip-12345"
         aria-controls="js-bolt-tooltip-12345"
    >
      <ssr-keep for="bolt-tooltip">
        CRM
      </ssr-keep>
    </div>
    <div id="js-bolt-tooltip-12345"
         class="c-bolt-tooltip__content"
         role="tooltip"
         aria-hidden="true"
    >
      <span class="c-bolt-tooltip__bubble">
        <ssr-keep for="bolt-tooltip">
          <span slot="content">
            Customer relationship management
          </span>
        </ssr-keep>
      </span>
    </div>
  </div>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: auto 1`] = `
<bolt-tooltip placement="auto"
              uuid="12345"
>
  <div class="c-bolt-tooltip  c-bolt-tooltip--text-wrap c-bolt-tooltip--text-align-center">
    <div role="button"
         tabindex="0"
         aria-describedby="js-bolt-tooltip-12345"
         aria-controls="js-bolt-tooltip-12345"
    >
      <ssr-keep for="bolt-tooltip">
        CRM
      </ssr-keep>
    </div>
    <div id="js-bolt-tooltip-12345"
         class="c-bolt-tooltip__content"
         role="tooltip"
         aria-hidden="true"
    >
      <span class="c-bolt-tooltip__bubble">
        <ssr-keep for="bolt-tooltip">
          <span slot="content">
            Customer relationship management
          </span>
        </ssr-keep>
      </span>
    </div>
  </div>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: bottom 1`] = `
<bolt-tooltip placement="bottom"
              uuid="12345"
>
  <div class="c-bolt-tooltip c-bolt-tooltip--bottom c-bolt-tooltip--text-wrap c-bolt-tooltip--text-align-center">
    <div role="button"
         tabindex="0"
         aria-describedby="js-bolt-tooltip-12345"
         aria-controls="js-bolt-tooltip-12345"
    >
      <ssr-keep for="bolt-tooltip">
        CRM
      </ssr-keep>
    </div>
    <div id="js-bolt-tooltip-12345"
         class="c-bolt-tooltip__content"
         role="tooltip"
         aria-hidden="true"
    >
      <span class="c-bolt-tooltip__bubble">
        <ssr-keep for="bolt-tooltip">
          <span slot="content">
            Customer relationship management
          </span>
        </ssr-keep>
      </span>
    </div>
  </div>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: bottom-end 1`] = `
<bolt-tooltip placement="bottom-end"
              uuid="12345"
>
  <div class="c-bolt-tooltip c-bolt-tooltip--bottom-end c-bolt-tooltip--text-wrap c-bolt-tooltip--text-align-center">
    <div role="button"
         tabindex="0"
         aria-describedby="js-bolt-tooltip-12345"
         aria-controls="js-bolt-tooltip-12345"
    >
      <ssr-keep for="bolt-tooltip">
        CRM
      </ssr-keep>
    </div>
    <div id="js-bolt-tooltip-12345"
         class="c-bolt-tooltip__content"
         role="tooltip"
         aria-hidden="true"
    >
      <span class="c-bolt-tooltip__bubble">
        <ssr-keep for="bolt-tooltip">
          <span slot="content">
            Customer relationship management
          </span>
        </ssr-keep>
      </span>
    </div>
  </div>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: bottom-start 1`] = `
<bolt-tooltip placement="bottom-start"
              uuid="12345"
>
  <div class="c-bolt-tooltip c-bolt-tooltip--bottom-start c-bolt-tooltip--text-wrap c-bolt-tooltip--text-align-center">
    <div role="button"
         tabindex="0"
         aria-describedby="js-bolt-tooltip-12345"
         aria-controls="js-bolt-tooltip-12345"
    >
      <ssr-keep for="bolt-tooltip">
        CRM
      </ssr-keep>
    </div>
    <div id="js-bolt-tooltip-12345"
         class="c-bolt-tooltip__content"
         role="tooltip"
         aria-hidden="true"
    >
      <span class="c-bolt-tooltip__bubble">
        <ssr-keep for="bolt-tooltip">
          <span slot="content">
            Customer relationship management
          </span>
        </ssr-keep>
      </span>
    </div>
  </div>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: left 1`] = `
<bolt-tooltip placement="left"
              uuid="12345"
>
  <div class="c-bolt-tooltip c-bolt-tooltip--left c-bolt-tooltip--text-wrap c-bolt-tooltip--text-align-center">
    <div role="button"
         tabindex="0"
         aria-describedby="js-bolt-tooltip-12345"
         aria-controls="js-bolt-tooltip-12345"
    >
      <ssr-keep for="bolt-tooltip">
        CRM
      </ssr-keep>
    </div>
    <div id="js-bolt-tooltip-12345"
         class="c-bolt-tooltip__content"
         role="tooltip"
         aria-hidden="true"
    >
      <span class="c-bolt-tooltip__bubble">
        <ssr-keep for="bolt-tooltip">
          <span slot="content">
            Customer relationship management
          </span>
        </ssr-keep>
      </span>
    </div>
  </div>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: left-end 1`] = `
<bolt-tooltip placement="left-end"
              uuid="12345"
>
  <div class="c-bolt-tooltip c-bolt-tooltip--left-end c-bolt-tooltip--text-wrap c-bolt-tooltip--text-align-center">
    <div role="button"
         tabindex="0"
         aria-describedby="js-bolt-tooltip-12345"
         aria-controls="js-bolt-tooltip-12345"
    >
      <ssr-keep for="bolt-tooltip">
        CRM
      </ssr-keep>
    </div>
    <div id="js-bolt-tooltip-12345"
         class="c-bolt-tooltip__content"
         role="tooltip"
         aria-hidden="true"
    >
      <span class="c-bolt-tooltip__bubble">
        <ssr-keep for="bolt-tooltip">
          <span slot="content">
            Customer relationship management
          </span>
        </ssr-keep>
      </span>
    </div>
  </div>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: left-start 1`] = `
<bolt-tooltip placement="left-start"
              uuid="12345"
>
  <div class="c-bolt-tooltip c-bolt-tooltip--left-start c-bolt-tooltip--text-wrap c-bolt-tooltip--text-align-center">
    <div role="button"
         tabindex="0"
         aria-describedby="js-bolt-tooltip-12345"
         aria-controls="js-bolt-tooltip-12345"
    >
      <ssr-keep for="bolt-tooltip">
        CRM
      </ssr-keep>
    </div>
    <div id="js-bolt-tooltip-12345"
         class="c-bolt-tooltip__content"
         role="tooltip"
         aria-hidden="true"
    >
      <span class="c-bolt-tooltip__bubble">
        <ssr-keep for="bolt-tooltip">
          <span slot="content">
            Customer relationship management
          </span>
        </ssr-keep>
      </span>
    </div>
  </div>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: right 1`] = `
<bolt-tooltip placement="right"
              uuid="12345"
>
  <div class="c-bolt-tooltip c-bolt-tooltip--right c-bolt-tooltip--text-wrap c-bolt-tooltip--text-align-center">
    <div role="button"
         tabindex="0"
         aria-describedby="js-bolt-tooltip-12345"
         aria-controls="js-bolt-tooltip-12345"
    >
      <ssr-keep for="bolt-tooltip">
        CRM
      </ssr-keep>
    </div>
    <div id="js-bolt-tooltip-12345"
         class="c-bolt-tooltip__content"
         role="tooltip"
         aria-hidden="true"
    >
      <span class="c-bolt-tooltip__bubble">
        <ssr-keep for="bolt-tooltip">
          <span slot="content">
            Customer relationship management
          </span>
        </ssr-keep>
      </span>
    </div>
  </div>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: right-end 1`] = `
<bolt-tooltip placement="right-end"
              uuid="12345"
>
  <div class="c-bolt-tooltip c-bolt-tooltip--right-end c-bolt-tooltip--text-wrap c-bolt-tooltip--text-align-center">
    <div role="button"
         tabindex="0"
         aria-describedby="js-bolt-tooltip-12345"
         aria-controls="js-bolt-tooltip-12345"
    >
      <ssr-keep for="bolt-tooltip">
        CRM
      </ssr-keep>
    </div>
    <div id="js-bolt-tooltip-12345"
         class="c-bolt-tooltip__content"
         role="tooltip"
         aria-hidden="true"
    >
      <span class="c-bolt-tooltip__bubble">
        <ssr-keep for="bolt-tooltip">
          <span slot="content">
            Customer relationship management
          </span>
        </ssr-keep>
      </span>
    </div>
  </div>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: right-start 1`] = `
<bolt-tooltip placement="right-start"
              uuid="12345"
>
  <div class="c-bolt-tooltip c-bolt-tooltip--right-start c-bolt-tooltip--text-wrap c-bolt-tooltip--text-align-center">
    <div role="button"
         tabindex="0"
         aria-describedby="js-bolt-tooltip-12345"
         aria-controls="js-bolt-tooltip-12345"
    >
      <ssr-keep for="bolt-tooltip">
        CRM
      </ssr-keep>
    </div>
    <div id="js-bolt-tooltip-12345"
         class="c-bolt-tooltip__content"
         role="tooltip"
         aria-hidden="true"
    >
      <span class="c-bolt-tooltip__bubble">
        <ssr-keep for="bolt-tooltip">
          <span slot="content">
            Customer relationship management
          </span>
        </ssr-keep>
      </span>
    </div>
  </div>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: top 1`] = `
<bolt-tooltip placement="top"
              uuid="12345"
>
  <div class="c-bolt-tooltip c-bolt-tooltip--top c-bolt-tooltip--text-wrap c-bolt-tooltip--text-align-center">
    <div role="button"
         tabindex="0"
         aria-describedby="js-bolt-tooltip-12345"
         aria-controls="js-bolt-tooltip-12345"
    >
      <ssr-keep for="bolt-tooltip">
        CRM
      </ssr-keep>
    </div>
    <div id="js-bolt-tooltip-12345"
         class="c-bolt-tooltip__content"
         role="tooltip"
         aria-hidden="true"
    >
      <span class="c-bolt-tooltip__bubble">
        <ssr-keep for="bolt-tooltip">
          <span slot="content">
            Customer relationship management
          </span>
        </ssr-keep>
      </span>
    </div>
  </div>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: top-end 1`] = `
<bolt-tooltip placement="top-end"
              uuid="12345"
>
  <div class="c-bolt-tooltip c-bolt-tooltip--top-end c-bolt-tooltip--text-wrap c-bolt-tooltip--text-align-center">
    <div role="button"
         tabindex="0"
         aria-describedby="js-bolt-tooltip-12345"
         aria-controls="js-bolt-tooltip-12345"
    >
      <ssr-keep for="bolt-tooltip">
        CRM
      </ssr-keep>
    </div>
    <div id="js-bolt-tooltip-12345"
         class="c-bolt-tooltip__content"
         role="tooltip"
         aria-hidden="true"
    >
      <span class="c-bolt-tooltip__bubble">
        <ssr-keep for="bolt-tooltip">
          <span slot="content">
            Customer relationship management
          </span>
        </ssr-keep>
      </span>
    </div>
  </div>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component Placement of the tooltip bubble: top-start 1`] = `
<bolt-tooltip placement="top-start"
              uuid="12345"
>
  <div class="c-bolt-tooltip c-bolt-tooltip--top-start c-bolt-tooltip--text-wrap c-bolt-tooltip--text-align-center">
    <div role="button"
         tabindex="0"
         aria-describedby="js-bolt-tooltip-12345"
         aria-controls="js-bolt-tooltip-12345"
    >
      <ssr-keep for="bolt-tooltip">
        CRM
      </ssr-keep>
    </div>
    <div id="js-bolt-tooltip-12345"
         class="c-bolt-tooltip__content"
         role="tooltip"
         aria-hidden="true"
    >
      <span class="c-bolt-tooltip__bubble">
        <ssr-keep for="bolt-tooltip">
          <span slot="content">
            Customer relationship management
          </span>
        </ssr-keep>
      </span>
    </div>
  </div>
</bolt-tooltip>
`;

exports[`<bolt-tooltip> Component UUID of the tooltip 1`] = `
<bolt-tooltip uuid="custom-unique-id">
  <div class="c-bolt-tooltip c-bolt-tooltip--top c-bolt-tooltip--text-wrap c-bolt-tooltip--text-align-center">
    <div role="button"
         tabindex="0"
         aria-describedby="js-bolt-tooltip-custom-unique-id"
         aria-controls="js-bolt-tooltip-custom-unique-id"
    >
      <ssr-keep for="bolt-tooltip">
        CRM
      </ssr-keep>
    </div>
    <div id="js-bolt-tooltip-custom-unique-id"
         class="c-bolt-tooltip__content"
         role="tooltip"
         aria-hidden="true"
    >
      <span class="c-bolt-tooltip__bubble">
        <ssr-keep for="bolt-tooltip">
          <span slot="content">
            Customer relationship management
          </span>
        </ssr-keep>
      </span>
    </div>
  </div>
</bolt-tooltip>
`;
